### PR TITLE
feat(material-experimental/mdc-slider): WIP implement the basics for …

### PR DESCRIPTION
**Pull request by wagnermaciel**
…mdc slider

### CODE REVIEW

**Comment by wagnermaciel** [comment]
I haven't started working on making this backwards compatible yet.

Also, this is a pretty massive change so I'm happy to stretch this into multiple commits or PRs or whatever ya'll prefer

File: src/dev-app/mdc-slider/mdc-slider-demo.html [hunk 1]
```diff
@@ -1,53 +1,20 @@
-<h1>Default Slider</h1>
-Label <mat-slider #slidey aria-label="Basic slider"></mat-slider>
-{{slidey.value}}
+<p>Continuous slider</p>
```
Comment (mmalerba) [inline src/dev-app/mdc-slider/mdc-slider-demo.html:1]: Usually we leave the mdc demo pretty much the same as the non-mdc demo. That way we can verify that all the same usages still work. Is there a reason you changed it so much here?

File: src/material-experimental/mdc-slider/slider-thumb.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<ng-container *ngIf="isDiscrete">
+	<div class="mdc-slider__value-indicator-container">
+		<div class="mdc-slider__value-indicator">
+			<span class="mdc-slider__value-indicator-text">
```
Comment (mmalerba) [inline src/material-experimental/mdc-slider/slider-thumb.html:4]: style nit: tabs

File: src/material-experimental/mdc-slider/module.ts [hunk 3]
```diff
@@ -10,11 +10,12 @@ import {CommonModule} from '@angular/common';
 import {NgModule} from '@angular/core';
 import {MatCommonModule} from '@angular/material-experimental/mdc-core';
 import {MatSlider} from './slider';
+import {MatSliderThumb} from './slider-thumb';
 
 @NgModule({
   imports: [MatCommonModule, CommonModule],
-  exports: [MatSlider, MatCommonModule],
-  declarations: [MatSlider],
+  exports: [MatSlider, MatSliderThumb, MatCommonModule],
```
Comment (mmalerba) [inline src/material-experimental/mdc-slider/module.ts:17]: Should we be exporting `MatSliderThumb` or is it just an internal thing?

File: src/material-experimental/mdc-slider/slider.ts [hunk 4]
```diff
@@ -87,67 +61,83 @@ export class MatSliderChange {
...
+    return this._isDiscrete;
+  }
+  set isDiscrete(value) {
+    this._isDiscrete = coerceBooleanProperty(value);
+  }
+  private _isDiscrete = false;
+
+  /** Whether this is a ranged slider. */
+  @Input()
+  get isRange(): boolean {
```
Comment (mmalerba) [inline src/material-experimental/mdc-slider/slider.ts:134]: I thought we were going to have a design doc about this before adding new stuff to the API?

File: src/dev-app/mdc-slider/mdc-slider-demo.html (hunk 1 above)
Comment (wagnermaciel) [reply src/dev-app/mdc-slider/mdc-slider-demo.html:1]: Yeah, I wanted to test each of the different ways the slider could be used. I'll revert this once I'm done testing. For now I'll add a todo to remember to do so

File: src/material-experimental/mdc-slider/module.ts (hunk 3 above)
Comment (wagnermaciel) [reply src/material-experimental/mdc-slider/module.ts:17]: It's internal. I still don't fully understand `NgModule`s so this is the result of me trying a bunch of stuff to get this to work

File: src/material-experimental/mdc-slider/slider.ts (hunk 4 above)
Comment (wagnermaciel) [reply src/material-experimental/mdc-slider/slider.ts:134]: We are. This current API that's implemented here is not the final one. This is as basic of an implementation I could create to get an understanding of how creating MDC components works

File: src/material-experimental/mdc-slider/slider.ts [hunk 5]
```diff
@@ -87,214 +65,264 @@ export class MatSliderChange {
...
-    registerEventHandler: () => {},
-    deregisterEventHandler: () => {},
-    registerThumbEventHandler: () => {},
-    deregisterThumbEventHandler: () => {},
-    registerBodyEventHandler: () => {},
-    deregisterBodyEventHandler: () => {},
-    registerWindowEventHandler: () => {},
-    deregisterWindowEventHandler: () => {},
-    removeTrackActiveStyleProperty: (_propertyName: string) => {},
+    hasClass: (_className: string) => {
```
Comment (jelbourn) [inline src/material-experimental/mdc-slider/slider.ts:174]: Can you extract this adapter out into a standalone class? See this PR for a reference https://github.com/angular/components/pull/19980

**Review by jelbourn — COMMENTED** [review]
I did a quick pass through for a first round of comments

**Pull request closed without merging**